planning/autoware_diffusion_planner/src/diffusion_planner_node.cpp

```diff
@@ -539,20 +579,46 @@ void DiffusionPlanner::publish_predictions(const std::vector<float> & prediction
 {
   constexpr int64_t batch_idx = 0;
   constexpr int64_t ego_agent_idx = 0;
-  auto output_trajectory = postprocess::create_trajectory(
+
+  const autoware_planning_msgs::msg::Trajectory output_trajectory = postprocess::create_trajectory(
     predictions, this->now(), transforms_.first, batch_idx, ego_agent_idx);
   pub_trajectory_->publish(output_trajectory);
 
-  auto ego_trajectory_as_candidate_msg = postprocess::to_candidate_trajectories_msg(
-    output_trajectory, generator_uuid_, "DiffusionPlanner");
+  // Publish all batch results as candidate trajectories
+  const int batch_size = params_.batch_size;
+
+  // Start with first trajectory as main candidate
+  autoware_internal_planning_msgs::msg::CandidateTrajectories ego_trajectory_as_candidate_msg =
```

Comment:
nits: Since [using autoware_planning_msgs::msg::Trajectory;](https://github.com/autowarefoundation/autoware_universe/blob/main/planning/autoware_diffusion_planner/include/autoware/diffusion_planner/diffusion_planner_node.hpp#L79) is declared in header file
```suggestion
  CandidateTrajectories ego_trajectory_as_candidate_msg =
```

Reply:
https://github.com/autowarefoundation/autoware_universe/pull/11197/commits/55204dfbd1c53a0635193616836856aba3e79e39